### DISCUSSION TITLE
SelectPicker / TagPicker defaultvalue using async data not working

### DISCUSSION THREAD

**Discussion by fankof** [rsuite/rsuite#2467] [Q&A]
Hey

I'm new in React and rsuitejs. I created a SelectPicker filling the data from an API. Everything works fine if its not defaultvalues. But when I try select by default one of the values, nothing is selected
- If the defaultvalues are loaded via useEffect or are 'hardcoded', no errors, but no option selected

For the TagPicker
- If the defaultvalues are 'hardcoded' and the data is consumed from an API and set via useState, throws error that the valuekey doesnt exists

Ex SelectPicker:
countriesData is set using a useState
`<SelectPicker data={countriesData} defaultValue={'57'} valueKey="Id" searchable={false} labelKey="Name" locale={selectLocale} style={{ width: 224 }} />`

Ex TagPicker:
<TagPicker data={rolesData} defaultValue={['1', '2']}
                                            labelKey={'RoleName'} valueKey={'RoleId'} style={{ width: 300 }}
                                            locale={selectLocale} .........

Update.tsx:241 
        
       Uncaught TypeError: Cannot read properties of undefined (reading 'RoleName')

**Comment by SevenOutman** [comment]
Looks like your `rolesData` array contained some items whose value is `undefined`.

**Comment by errand** [comment]
Unfortunatelly i'm issuing the same behaviour
I have about 1000 tags, so i use async fetching. It works fine when creating new entities, but when edit, the selected Tags are not displaying until i start searching them by typing. Even then input contains not all saved, but only those that meet the search requirements.

I screened short video of the problem [here](https://www.veed.io/view/54a85de0-bee6-451d-b254-63275cbbaa95?panel=share)

**Reply by SevenOutman** [reply]
Could you provide a CodeSandbox demo so that I know what's the props like? You could use this template https://codesandbox.io/s/rsuite-template-jdydg7

**Comment by errand** [comment]
Hey @SevenOutman, thank you for your reply, 
actually, i must admit that i wasn't very attentive and lost the view of the defaultValues.
Iit works as designed. Thank you

**Comment by errand** [comment]
@SevenOutman , sorry, but i'm still here with common issue.
Then i use defaultValues it is not selected. If you click on select, you'll see the default value in dropdown, but i'm issuing selecting it by default
https://codesandbox.io/s/suspicious-sea-5g8872?file=/src/App.js

**Reply by SevenOutman** [reply]
There're two reasons your demo is not working as expected.

1. When `value` is provided, `defaultValue` won't take effect
2. You set `valueKey="id"` and `defaultValue={["errand"]}` but there's no item in the data array whose `id` is `"errand"`. Did you mean to set `valueKey="login"` or `defaultValue={[1334432]}`?

**Comment by errand** [comment]
@SevenOutman , thank you!
